### DISCUSSION TITLE
Is there a way to see if a pgroonga index is healthy ?

### DISCUSSION THREAD

**Discussion by ghevge** [pgroonga/pgroonga#411] [Q&A]
I'm mainly looking for a query (preferably a SELECT or any other query that doesn't lock tables) that will be able to tell me what is the health of a pgroonga indexes on a given table.

Is there such a query available ?

Thanks

**Comment by kou** [comment]
PGroonga doesn't provide a function for it for now.
But we'll add a function that can be used for it in a few months.

**Comment by ghevge** [comment]
That's a pity. Without such a function I will have to manually stop my app, before any index can be recreated, in case of a failure.

**Comment by kou** [comment]
I think that you need to do it even when we have such a function on your system (#397).
Because your PostgreSQL is always crashed when your system is shutdown.
